### PR TITLE
Use regular CREATE TABLE instead of CREATE TEMPORARY

### LINKED ISSUE #30631: Upgrade to 25 throws: Statement violates GTID consistency
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

After upgrading to v25 I got the following error on logs:

```
Caused by: liquibase.exception.DatabaseException: Statement violates GTID consistency: CREATE TEMPORARY TABLE and DROP TEMPORARY TABLE can only be executed outside transactional context.  These statements are also not allowed in a function or trigger because functions and triggers are also considered to be multi-statement transactions. [Failed SQL: (1787) CREATE TEMPORARY TABLE TEMP_USER_CONSENT_IDS AS SELECT uc.ID FROM keycloak.USER_CONSENT uc INNER JOIN ( SELECT CLIENT_ID, USER_ID, MAX(LAST_UPDATED_DATE) AS MAX_UPDATED_DATE FROM keycloak.USER_CONSENT GROUP BY CLIENT_ID, USER_ID HAVING COUNT(*) > 1 ) max_dates ON uc.CLIENT_ID = max_dates.CLIENT_ID AND uc.USER_ID = max_dates.USER_ID AND uc.LAST_UPDATED_DATE = max_dates.MAX_UPDATED_DATE]
```

### Version

25.0.0

### Regression

- [X] The issue is a regression

### Expected behavior

startup should working fine post any db migrations/validations

### Actual behavior

statup fails

### How to Reproduce?

Using MySQL upgrade to v25

### Anything else?

I assume this is a issue with mysql alone since I don't think GTID is a thing on the others, and we could temporarily disable GTID consistency, but its probably best KC does this is a way that does need the work around.

### PR DESCRIPTION
**Pull request by mike-pt**
Some hosted/managed environments like google CloudSQL, might not support this type of statement (i.e. when using replication and GTID) as seen in https://github.com/keycloak/keycloak/issues/30631

Since we are dropping the table anyway it seems a regular CREATE statement should work fine here.

Closes #30631

Signed-off-by: mike-pt <mike-pt@users.noreply.github.com>

### CODE REVIEW

**Comment by mike-pt** [comment]
rebased with main

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ protected void generateStatementsImpl() throws CustomChangeException {
                         " AND uc.USER_ID = max_dates.USER_ID AND uc.LAST_UPDATED_DATE = max_dates.MAX_UPDATED_DATE)"
         ));
         statements.add(new RawSqlStatement(
-                "  CREATE TEMPORARY TABLE TEMP_USER_CONSENT_IDS" +
+                "  CREATE TABLE TEMP_USER_CONSENT_IDS" +
```
Comment (pedroigor) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java:38]: The corresponding statement to drop the table is still `DROP TEMPORARY TABLE IF EXISTS TEMP_USER_CONSENT_IDS`. Shouldn't it be `DROP TABLE IF EXISTS TEMP_USER_CONSENT_IDS`?

Similar to what you did in the following change.

**Comment by pedroigor** [comment]
@mike-pt Please, see https://github.com/keycloak/keycloak/actions/runs/9645408727/job/26890406244?pr=30669#step:4:2086.

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ protected void generateStatementsImpl() throws CustomChangeException {
         ));
         statements.add(new DeleteStatement(null, null, database.correctObjectName("USER_CONSENT", Table.class))
                 .setWhere("ID IN (SELECT ID FROM TEMP_USER_CONSENT_IDS2)"));
-        statements.add(new RawSqlStatement("DROP TEMPORARY TABLE IF EXISTS TEMP_USER_CONSENT_IDS2"));
+        statements.add(new RawSqlStatement("DROP TABLE IF EXISTS TEMP_USER_CONSENT_IDS2"));
```
Comment (mc-zact) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java:67]: @pedroigor  isn't that here? ^

**Comment by mike-pt** [comment]
> @mike-pt Please, see https://github.com/keycloak/keycloak/actions/runs/9645408727/job/26890406244?pr=30669#step:4:2086.

I see the issue now, I missed one more DROP statement still using TEMPORARY.
Should be good now

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java (hunk 1 above)
Comment (mike-pt) [reply model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate25_0_0_MySQL_ConsentConstraints.java:38]: Yeah I missed one :D

**Comment by mike-pt** [comment]
@pedroigor  is there any other pending action from me on this?

**Comment by codespearhead** [comment]
It sounds like a reasonable fix.

Could you remove the merge commit to maintain a linear Git history?

To do so, run:

```bash
git clone --single-branch --branch patch-1 git@github.com:mike-pt/keycloak.git
cd keycloak

# Remove the merge commit in the local repository
git reset --hard HEAD~1

# Rebase onto main
git remote add upstream git@github.com:keycloak/keycloak.git
git pull --rebase upstream main

# Make sure the commit is signed
git commit --amend --signoff

# Force push to remove the previously pushed merge commit from your fork by resetting the remote repository to match the local repository
git push --force
```

**Comment by mike-pt** [comment]
@codespearhead recommendations addressed

**Review by codespearhead — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
LGTM, thank you for the change. 
@pedroigor - are you ok if we merge this?

**Comment by mike-pt** [comment]
@pedroigor  is there anything blocking this on my side?

**Comment by mike-pt** [comment]
@pedroigor sorry for the re ping but its been a month since my last comment and I'm not sure if there is something blocking this or now? It seems it was already approved just needs to be merged in?

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
Sorry for the delay. Thanks!

**Pull request merged**